app/controllers/tagged_courses_controller.rb

```diff
@@ -41,9 +42,21 @@ def set_tag
 
   def set_courses_and_presenter
     @courses = Tag.courses_tagged_with(@tag)
-    # This fake Campaign is so that we can reuse campaigns views
-    @campaign = Campaign.new(slug: @tag, title: "Tag: #{@tag}")
     @presenter = CoursesPresenter.new(current_user:, tag: @tag,
                                       courses_list: @courses, page: @page)
   end
+
+  # Loads CoursesUsers records with role 4 and filters by wiki experts, avoiding N+1 queries
+  def load_wiki_experts
+    return @wiki_experts if @wiki_experts # Avoid re-loading if already loaded
+
+    course_ids = @courses.pluck(:id)
+    wiki_experts_set = SpecialUsers.special_users[:wikipedia_experts]&.to_set
+
+    @wiki_experts = CoursesUsers
+                    .where(course_id: course_ids, role: 4)
+                    .includes(:user)
+                    .select { |course_user| wiki_experts_set.include?(course_user.user.username) }
```

Comment:
I think this could be done a little bit more efficiently, by using SpecialUsers.wikipedia_experts to the user records.

There's also no need to pluck IDs for the courses; you can instead do `CoursesUsers.where(course: @courses, user: SpecialUsers.wikipedia_experts, role: ROLE)`